src/core/src/op/mod.cpp

```diff
@@ -4,13 +4,30 @@
 
 #include "openvino/op/mod.hpp"
 
+#include "bound_evaluate.hpp"
 #include "element_visitor.hpp"
 #include "itt.hpp"
-#include "openvino/core/shape_util.hpp"
+#include "openvino/core/validation_util.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/equal.hpp"
+#include "openvino/op/logical_or.hpp"
+#include "openvino/op/select.hpp"
 #include "openvino/reference/mod.hpp"
 #include "utils.hpp"
+#include "validation_util.hpp"
 
 namespace ov {
+namespace util {
+namespace {
+Tensor make_tensor_of_value(const element::Type_t et, const int64_t value) {
+    auto c = op::v0::Constant(et, Shape{}, value);
+    auto t = Tensor(et, Shape{});
+    std::memcpy(t.data(), c.get_data_ptr(), t.get_byte_size());
+    return t;
```

Comment:
Because `Constant` convert input value correctly to given precision and prepares correctly the storage. 
Without out the storage has to be created manually (depends on precision -> `switch case` etc) and then copied.